src/main/java/com/apifest/oauth20/tests/RefreshTokenFlowTest.java

```diff
@@ -52,23 +52,24 @@ public void when_refresh_token_valid_then_access_token_issued() throws Exception
         assertTrue(!newAccessToken.contains("{\"error\""));
     }
 
-    @Test
-    public void when_obtain_access_token_with_already_used_refresh_token_return_error() throws Exception {
-        // GIVEN
-        String authCode = obtainAuthCode(clientId, DEFAULT_REDIRECT_URI);
-
-        // WHEN
-        String accessTokenResponse = obtainAccessTokenResponse("authorization_code", authCode, clientId, clientSecret, DEFAULT_REDIRECT_URI);
-        String refreshToken = extractRefreshToken(accessTokenResponse);
-        String newAccessToken = obtainAccessTokenByRefreshToken("refresh_token", refreshToken, clientId, clientSecret, DEFAULT_SCOPE);
-        String superNewAccessToken = obtainAccessTokenByRefreshToken("refresh_token", refreshToken, clientId, clientSecret, DEFAULT_SCOPE);
-
-        // THEN
-        assertNotNull(newAccessToken);
-        assertTrue(!newAccessToken.contains("{\"error\""));
-        assertNotNull(superNewAccessToken);
-        assertTrue(superNewAccessToken.contains("{\"error\""));
-    }
+//    @Test
```

Comment:
You are right the test is not passing no matter which DB is used. It seems it is an old test that was written to an old functionality of the ApiFest OAuth 2.0 Server. Right now a refresh token is not invalidated if it used more than once as the refresh token of the new access token generated is the same. Only the old access token is invalidated. A refresh token cannot be used anymore if it has expired. Thanks for you code review. Please, remove the test instead of commenting it.